mcorpc/authz_rego.go

```diff
@@ -0,0 +1,149 @@
+package mcorpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/choria-io/go-choria/choria"
+	"github.com/choria-io/go-config"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/sirupsen/logrus"
+)
+
+type regoMatcher interface {
+	Set(caller string, actions string, facts string, classes string)
+}
+
+type regoPolicy struct {
+	cfg     *config.Config
+	req     *Request
+	agent   *Agent
+	log     *logrus.Entry
+	matcher regoMatcher
+}
+
+type regoPolicyPolicy struct {
+	caller  string
+	actions string
+	facts   string
+	classes string
+	log     *logrus.Entry
+}
+
+func (r *regoPolicyPolicy) Set(caller string, actions string, facts string, classes string) {
```

Comment:
I don't think I actually use this code.

Reply:
yup, whole struct and all its uses can go, the action policy one was quite complex so i split it to make testing easier, but this one is not so dont need the extra bit